### PR TITLE
[Quant][fx] Lower reference conv[1-3]d module

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #69228

Implement lowering logic for reference conv modules,
similar to https://github.com/pytorch/pytorch/pull/65723.

Differential Revision: [D32890743](https://our.internmc.facebook.com/intern/diff/D32890743/)

Differential Revision: [D32890743](https://our.internmc.facebook.com/intern/diff/D32890743)

### CODE REVIEW

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 1]
```diff
@@ -35,31 +48,27 @@ def _lower_ref_linear_module(model: QuantizedGraphModule) -> QuantizedGraphModul
             continue
 
         # this can be removed if we add support for constants in is_match
-        if dtype != torch.quint8:
+        if q_class == torch.nn.quantized.Linear and dtype != torch.quint8:
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:51]: this check is valid for both linear and conv I think

File: test/quantization/fx/test_quantize_fx.py [hunk 2]
```diff
@@ -3199,6 +3199,38 @@ def forward(self, x):
         # checking result match
         self.assertEqual(out_ref, out)
 
+    def test_conv_lowering(self):
+        convs = [nn.Conv1d, nn.Conv2d, nn.Conv3d]
```
Comment (jerryzh168) [inline test/quantization/fx/test_quantize_fx.py:3203]: nit: we typically use a map from dim to conv class, e.g.: https://github.com/pytorch/pytorch/blob/master/test/quantization/fx/test_quantize_fx.py#L3619, that might be clearer

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 3]
```diff
@@ -6,13 +6,26 @@
...
-        MatchAllNode, MatchAllNode, MatchAllNode)
+# Mapping from reference module class to the replacement quantized module class for lowering
+_lower_module_map = {
+    torch.nn.quantized._reference.Linear: torch.nn.quantized.Linear,
+    torch.nn.quantized._reference.Conv1d: torch.nn.quantized.Conv1d,
+    torch.nn.quantized._reference.Conv2d: torch.nn.quantized.Conv2d,
+    torch.nn.quantized._reference.Conv3d: torch.nn.quantized.Conv3d,
+}
+
+def _lower_ref(model: QuantizedGraphModule, ref_class: type) -> QuantizedGraphModule:
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:17]: nit: maybe rename to `_lower_weighted_ref_module` to be more explicit

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 4]
```diff
@@ -6,13 +6,26 @@
...
 
-def _lower_ref_linear_module(model: QuantizedGraphModule) -> QuantizedGraphModule:
-    # traverse the graph and find dequantize - ref quantized linear - quantize patterns and
-    # and replace it with quantized linear modules
-    pattern = (
-        torch.quantize_per_tensor,
-        (torch.nn.quantized._reference.Linear, "dequantize"),
-        MatchAllNode, MatchAllNode, MatchAllNode)
+# Mapping from reference module class to the replacement quantized module class for lowering
+_lower_module_map = {
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:10]: nit: we typically use something like `LOWER_MODULE_MAP` to indicate this is a constant map

**Review by jerryzh168 — APPROVED** [review]
Looks great!

**Comment by jerryzh168** [comment]
please import the PR to phabricator with ghimport as well

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32890743).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32890743).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32890743).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32890743).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32890743).
